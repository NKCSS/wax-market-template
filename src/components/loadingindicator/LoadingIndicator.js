import React from 'react'

export const LoadingIndicator = (props) => {
    const text = props.text ?? props.children ?? null
    return text ? (
        <div>
            <div className="flex justify-center m-auto">
                <img
                    data-testid="loading-spinner"
                    src="/Loader.svg"
                    className="w-8 h-8 m-auto animate-spin duration-1000"
                    alt="Loading"
                />
            </div>
            <div data-testid="loading-text">{text}</div>
        </div>
    ) : (
        <div className="flex justify-center m-auto">
            <img
                data-testid="loading-spinner"
                src="/Loader.svg"
                className="w-8 h-8 m-auto animate-spin duration-1000"
                alt="Loading"
            />
        </div>
    )
}

export default LoadingIndicator
