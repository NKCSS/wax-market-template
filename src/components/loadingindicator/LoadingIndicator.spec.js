import { render, screen } from '@testing-library/react'
import { LoadingIndicator } from './LoadingIndicator'

describe('LoadingIndicator', () => {
    it('should render', () => {
        const { container } = render(<LoadingIndicator />)
        expect(container).toMatchSnapshot()
    })
    it('should render without text', () => {
        render(<LoadingIndicator></LoadingIndicator>)
        expect(screen.getByTestId('loading-spinner')).toBeInTheDocument()
        expect(screen.queryByTestId('loading-text')).not.toBeInTheDocument()
    })
    it('should render with text', () => {
        render(<LoadingIndicator>Hello World!</LoadingIndicator>)
        expect(screen.getByTestId('loading-spinner')).toBeInTheDocument()
        expect(screen.getByTestId('loading-text')).toBeInTheDocument()
    })
    it('should render with text property (@deprecated)', () => {
        render(<LoadingIndicator text="Hello World!" />)
        expect(screen.getByTestId('loading-spinner')).toBeInTheDocument()
        expect(screen.getByTestId('loading-text')).toBeInTheDocument()
    })
})
