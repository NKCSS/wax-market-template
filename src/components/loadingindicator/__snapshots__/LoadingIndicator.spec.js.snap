// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator should render 1`] = `
<div>
  <div
    class="flex justify-center m-auto"
  >
    <img
      alt="Loading"
      class="w-8 h-8 m-auto animate-spin duration-1000"
      data-testid="loading-spinner"
      src="/Loader.svg"
    />
  </div>
</div>
`;
